openedx/features/qverse_features/registration/signals.py

```diff
@@ -1,20 +1,187 @@
+    else:
+        LOGGER.info('{} edx profile has been updated.'.format(edx_user.username))
+
+
+def _create_or_update_qverse_user_profile(edx_user, user_info):
+    """
+    Creates/Updates qverse user profile.
+
+    It also stores the status of the creation/update in the receiving list which
+    contains the content which will be written on output file.
+
+    Arguments:
+        edx_user (User): An edx user instance whose qverse profile is going to be created/updated
+        user_info (dict): User info whose profile is going to be created/updated
+    """
+    department = None
+    try:
+        department = Department.objects.get(number=user_info.get('departmentid'))
+    except Department.DoesNotExist:
+        LOGGER.exception('The Department with id {} does not exist.'.format(user_info.get('departmentid')))
+        user_info.update({'status': 'Failed'})
+        raise ValueError('Invalid Department ID')
+
+    qverse_profile_data = {
+                            'current_level': user_info.get('levelid'),
+                            'other_name': user_info.get('othername'),
+                            'mobile_number': user_info.get('mobile'),
+                            'programme': user_info.get('programmeid'),
+                            'department': department,
+                            'registration_number': user_info.get('regno')
+                            }
+    _, created = QVerseUserProfile.objects.update_or_create(user=edx_user, defaults=qverse_profile_data)
+    if created:
+        LOGGER.info('{} qverse profile has been created.'.format(edx_user.username))
+        user_info.update({'status': 'Created'})
+    else:
+        LOGGER.info('{} qverse profile has been updated.'.format(edx_user.username))
+        user_info.update({'status': 'Updated'})
+
+
+def _write_status_on_csv_file(filename, output_file_rows):
+    """
+    Writes on the given file.
+
+    Arguments:
+        filename (str): Complete path of the output file
+        output_file_rows (list): List of CSV file rows
+    """
+    try:
+        with open(filename, 'w') as csv_file:
+            if output_file_rows:
+                writer = DictWriter(csv_file, fieldnames=output_file_rows[0].keys())
+                writer.writeheader()
+                for row in output_file_rows:
+                    writer.writerow(row)
+    except IOError as error:
+        LOGGER.error('(filename) --- {}'.format(filename, error.strerror))
+
+
+def _validate_single_csv_row(row):
+    """
+    Validates that row contains certian values for required fields.
+
+    Arguments:
+        row (dict): Contains the single row of data from CSV file
+    """
+    OPTIONAL_FIELDS = [
+        'mobile', 'othername', 'status'
+    ]
+    required_values = [value for (key, value) in row.items() if key not in OPTIONAL_FIELDS]
+    if all(required_values):
+        return True
+    else:
+        raise ValueError()
```

Comment:
why this?

Reply:
This exception will be handled above and a comprehensive log message will be printed.

Reply:
removed as you said earlier.